doc/fpmsyncd/frr_sonic_protobuf_channel.md

```diff
@@ -0,0 +1,232 @@
+* SONiC fpmsyncd to handle Protobuf messages and Netlink messages received from FRR
+* SONiC fpmsyncd to SET/DEL entries (such as regular routes, SRv6 SIDs and SRv6 routes) to APPL_DB
+
+## 6. Architecture Design
+
+Currently, the communication channel between FRR and SONiC exclusively supports the transmission of information encoded in the Netlink format. As previously mentioned (refer to [Section 3.1](#31-context)), the Netlink format is tailored to meet the specifications and data model of the kernel. Consequently, it exhibits limitations in accommodating various use cases on the SONiC dataplane.
+
+To overcome the limitations imposed by the Netlink format, we propose extending the communication channel between FRR and SONiC. This extension would introduce the capability to transmit information encoded in a format independent on the kernel data model, specifically the Protobuf message format.
+
+This proposal introduces a new FPM module in FRR, named `dplane_fpm_pb`. This module enables FRR to encode information using Protobuf as the message format and subsequently transmit the Protobuf message to SONiC.
+
+On the SONiC side, we have a small daemon (`fpmsyncd`) that receives information from FRR and process it. We make some modifications to SONiC's `fpmsyncd`, allowing it to receive Protobuf messages from the new `dplane_fpm_pb` module, process them, and subsequently write the necessary information (such as SRv6 routes and SIDs) into the APPL_DB.
+
+The following figure shows the changes to the SONiC Architecture:
+
+![FRR-SONiC Protobuf Communication Channel - High-Level Architecture](images/fpm-pb-channel-architecture.png)	
+*Figure 2: FRR-SONiC Protobuf Communication Channel - High-Level Architecture*
+
+The new modules are colored in <span style="color:blue">blue</span>; the modified components are colored in <span style="color:green">green</span>.
+
+## 7. High-Level Design 
+
+### 7.1. FRR changes
+
+FRR has a module called `dplane_fpm_nl` which is responsible for exporting the routing information to SONiC. The routing information exported by `dplane_fpm_nl` is encoded using Netlink as the message format.
+
+In order to support transmitting information encoded in Protobuf format, we introduce a new module in FRR. The new module is called `dplane_fpm_pb` and is responsible for encoding and sending information to SONiC using Protobuf as the message format. The new FPM module (`dplane_fpm_pb`) is designed to coexist with the existing one (`dplane_fpm_nl`). The two modules can be run simultaneously.
+
+To enable the FPM Protobuf module in FRR, two steps are required:
+
+* FRR must be compiled with the support for the FPM Protobuf module.
+* The FRR zebra daemon must load the FPM Protobuf module during startup.
+
+To implement these modifications, a PR has been submitted to the FRR repository and is currently under review: https://github.com/FRRouting/frr/pull/14173.
+
```

Comment:
These changes are not yet merged to FRR. Could you provide the targeted FRR release with protobuf change?